platforms/core-runtime/build-configuration/src/main/java/org/gradle/internal/buildconfiguration/resolvers/ToolchainSupportedPlatformsMatrix.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.buildconfiguration.resolvers;
+
+import org.gradle.platform.Architecture;
+import org.gradle.platform.BuildPlatform;
+import org.gradle.platform.OperatingSystem;
+import org.gradle.platform.internal.DefaultBuildPlatform;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ToolchainSupportedPlatformsMatrix {
```

Comment:
❓ Check if this is still required

Reply:
This class is responsible of generating the supported list of `BuildPlatform` which later is used to create the `DefaultJavaToolchainRequest` to resolve the download URL from project applied `JavaToolchainRepository` and end up populating the `gradle/gradle-daemon-jvm.properties` with it. So if we want to continue generating all download URLs for all supported `BuildPlatform` then we do need to have something similar in place. 

_Note: by default the `32-bit` architecture is discarded_

Reply:
However, since the logic was moved directly to [UpdateDaemonJvm.java](https://github.com/gradle/gradle/blob/03741e0b25d860f0a677d1896b01db68a5baf624/platforms/core-runtime/build-configuration/src/main/java/org/gradle/buildconfiguration/tasks/UpdateDaemonJvm.java#L82) we can potentially remove it